docs/Setup/NodeSDK-setup.md

```diff
@@ -0,0 +1,253 @@
+
+### Chaincode Deployment Directory Structure
+
+To have the chaincode deployment succeed in network mode, you must properly set up the chaincode project outside of your Hyperledger Fabric source tree. These instructions will demonstrate how to properly set up the directory structure to deploy *chaincode_example02* in network mode.
+
+The chaincode project must be placed inside the `src` directory in your local `$GOPATH`. For example, the [chaincode_example02](https://github.com/hyperledger/fabric/blob/master/examples/chaincode/go/chaincode_example02/chaincode_example02.go) project may be placed inside `$GOPATH/src/` as shown below.
+
+```
+mkdir -p $GOPATH/src/github.com/chaincode_example02/
+cd $GOPATH/src/github.com/chaincode_example02
+curl GET https://raw.githubusercontent.com/hyperledger/fabric/master/examples/chaincode/go/chaincode_example02/chaincode_example02.go > chaincode_example02.go
+```
+
+Once you have placed your chaincode project inside the `src` directory in your local `$GOPATH`, you will need to vendor the dependencies. In your chaincode source directory, run the following commands:
+
+```
+go get -u github.com/kardianos/govendor
+cd $GOPATH/src/github.com/chaincode_example02
+govendor init
+govendor fetch github.com/hyperledger/fabric
+```
+
+Now, execute `go build` to verify that all of the chaincode dependencies are present.
+
+```
+go build
+```
+
+Next, we will switch over to the node sdk directory in the fabric repo to run the node sdk tests, to make sure you have everything properly set up. Verify that the [chain-tests.js](https://github.com/hyperledger/fabric/blob/master/sdk/node/test/unit/chain-tests.js) unit test file points to the correct chaincode project path. The default directory is set to `github.com/chaincode_example02/` as shown below. If you placed the sample chaincode elsewhere, then you will need to change that.
+
+```
+// Path to the local directory containing the chaincode project under $GOPATH
+var testChaincodePath = "github.com/chaincode_example02/";
+```
+
+**Note:** You will need to run `npm install` the first time you run the sdk tests, to install all of the dependencies. Set the `DEPLOY_MODE` environment variable to `net` and run the chain-tests as follows:
+
+```
+cd $GOPATH/src/github.com/hyperledger/fabric/sdk/node
+npm install
+export DEPLOY_MODE='net'
+node test/unit/chain-tests.js | node_modules/.bin/tap-spec
+```
+
+### Enabling TLS
+
+If you wish to activate TLS connection with the member services the following actions are needed:
+
+- Modify `$GOPATH/src/github.com/hyperledger/fabric/membersrvc/membersrvc.yaml` as follows:
+
+```
+server:
+     tls:
+        cert:
+            file: "/var/hyperledger/production/.membersrvc/tlsca.cert"
+        key:
+            file: "/var/hyperledger/production/.membersrvc/tlsca.priv"
+```
+
+This is needed to instruct the member services on which tls cert and key to use.
```

Comment:
Specifying to the Membership Services (TLS) Certificate Authority (TLSCA) what X.509 v3 Certificate (with a  corresponding Private Key) to use.